bilge-impl/src/from_bits.rs

```diff
@@ -44,24 +44,26 @@ fn analyze_enum(variants: Iter<Variant>, name: &Ident, internal_bitsize: BitSize
 
     variants.map(|variant| {
         let variant_name = &variant.ident;
-        let variant_value = value_assigner.assign(variant);
+        let variant_value = value_assigner.assign_unsuffixed(variant);
 
-        // might be useful for not generating "1u128 -> Self::Variant"
-        let variant_value: Expr = syn::parse_str(&variant_value.to_string()).unwrap_or_else(unreachable);
-
-        let from_int_match_arm = quote! {
-            #variant_value => Self::#variant_name,
+        let from_int_match_arm = if is_ident_of_fallback(fallback, variant_name) {
```

Comment:
Sure. 
I went for a free function because Rust makes this awkward with `Option<Fallback>` and no `let_chains`... but I forgot we can do this with a `match` statement.

Reply:
Ah, it's `Option<self>`... not sure how we can do this exactly then, but you seem to have an idea so go at it haha